test/server/server_test.cc

```diff
@@ -334,6 +334,25 @@ TEST_P(ServerInstanceImplTest, StatsFlushWhenServerIsStillInitializing) {
   server_thread->join();
 }
 
+// Validates that server stats are not flushed via timer when flush_stats_on_admin_access is set to
```

Comment:
Are there tests for whether flushing on admin access does what is expected? I'm actually a little unclear from reading this PR whether the flush effect is guaranteed to take effect by time you are reading from admin.

Reply:
It does take affect because we are calling flushStats on admin stats handler. However, there is no test in place for verifying that part yet. I am wondering we should add a case in admin_integration_test for it? Or any other simpler options?